plugins/module_utils/ansible_freeipa_module.py

```diff
@@ -691,6 +720,17 @@ def get_ipa_command_args(self, **kwargs):
 
             return args
 
+        def check_invalid_params(self, invalid):
```

Comment:
This method is not really doing a check of the parameters. The name is not matching the code. `fail_for_invalid_params_with_state` is too long, but there should be a better name.